lib/logger.js

```diff
@@ -825,33 +855,36 @@ Object.defineProperties(Logger.prototype, {
 /*
  *  Populate short-hand for each supported Log Level
  */
-for (const level of constants.LOG_LEVELS) {
-  const levelLowerCase = level.toLowerCase()
-
-  Object.defineProperties(Logger.prototype, {
-    [levelLowerCase]: {
-      enumerable: true
-    , writable: false
-    , configurable: false
-    , value(statement, opts) {
-        const optsType = typeOf(opts)
-        if (opts && optsType !== 'object') {
-          const err = new TypeError(
-            `If passed, log.${levelLowerCase} requires 'options' to be an object`
-          )
-          err.meta = {
-            got: optsType
+function createConvenienceMethods(levels, instance) {
+  for (const level of levels) {
+    if (!constants.LOG_LEVELS.includes(level)) continue
+    const levelLowerCase = level.toLowerCase()
+
+    Object.defineProperties(instance, {
+      [levelLowerCase]: {
+        enumerable: true
+      , writable: false
+      , configurable: false
+      , value(statement, opts) {
+          const optsType = typeOf(opts)
+          if (opts && optsType !== 'object') {
```

Comment:
the first check for opts isn't necessary here. its either an object or it isn't.

Reply:
🤦🏻

Reply:
Oh right.  The first branch is not for the type, it's for if a value is passed at all.  The parameter is optional, and if passed, must be an object.  `undefined` is ok, hence the first check.